sdk/azcore/internal/exported/request.go

```diff
@@ -67,6 +68,42 @@ func (ov opValues) get(value any) bool {
 	return ok
 }
 
+// NewRequestFromRequest creates a new policy.Request with an existing *http.Request
+// Exported as runtime.NewRequestFromRequest().
+func NewRequestFromRequest(req *http.Request) (*Request, error) {
+	policyReq := &Request{req: req}
+
+	if req.Body != nil {
+		// we can avoid a body copy here if the underlying stream is already a
+		// ReadSeekCloser.
+		readSeekCloser, isReadSeekCloser := req.Body.(io.ReadSeekCloser)
```

Comment:
Unfortunately this doesn't have the effect I was hoping for. `io.NopCloser` erases the `io.Seeker` which is likely the common case for requests created outside our SDKs.

Reply:
Doesn't hurt to leave it but we won't avoid copying as much as I'd hoped.

Reply:
That's okay, it's still instructive and anything we can do to shave off obvious stuff is still nice.